src/db_options.rs

```diff
@@ -2108,6 +2716,237 @@ pub enum DBRecoveryMode {
+        }
+    }
+
+    /// Sets the maximum number of files in a single compaction run.
+    ///
+    /// Default: UINT_MAX
+    pub fn set_max_merge_width(&mut self, num: c_int) {
+        unsafe {
+            ffi::rocksdb_universal_compaction_options_set_max_merge_width(self.inner, num);
+        }
+    }
+
+    /// sets the size amplification.
+    ///
+    /// It is defined as the amount (in percentage) of
+    /// additional storage needed to store a single byte of data in the database.
+    /// For example, a size amplification of 2% means that a database that
+    /// contains 100 bytes of user-data may occupy upto 102 bytes of
+    /// physical storage. By this definition, a fully compacted database has
+    /// a size amplification of 0%. Rocksdb uses the following heuristic
+    /// to calculate size amplification: it assumes that all files excluding
+    /// the earliest file contribute to the size amplification.
+    ///
+    /// Default: 200, which means that a 100 byte database could require upto 300 bytes of storage.
+    pub fn set_max_size_amplification_percent(&mut self, v: c_int) {
+        unsafe {
+            ffi::rocksdb_universal_compaction_options_set_max_size_amplification_percent(
+                self.inner, v,
+            );
+        }
+    }
+
+    /// Sets the percentage of compression size.
+    ///
+    /// If this option is set to be -1, all the output files
+    /// will follow compression type specified.
+    ///
+    /// If this option is not negative, we will try to make sure compressed
+    /// size is just above this value. In normal cases, at least this percentage
+    /// of data will be compressed.
+    /// When we are compacting to a new file, here is the criteria whether
+    /// it needs to be compressed: assuming here are the list of files sorted
+    /// by generation time:
+    ///    A1...An B1...Bm C1...Ct
+    /// where A1 is the newest and Ct is the oldest, and we are going to compact
+    /// B1...Bm, we calculate the total size of all the files as total_size, as
+    /// well as  the total size of C1...Ct as total_C, the compaction output file
+    /// will be compressed iff
+    ///   total_C / total_size < this percentage
+    ///
+    /// Default: -1
+    pub fn set_compression_size_percent(&mut self, v: c_int) {
+        unsafe {
+            ffi::rocksdb_universal_compaction_options_set_compression_size_percent(self.inner, v);
+        }
+    }
+
+    /// Sets the algorithm used to stop picking files into a single compaction run.
+    ///
+    /// Default: ::Total
+    pub fn set_stop_style(&mut self, style: UniversalCompactionStopStyle) {
+        unsafe {
+            ffi::rocksdb_universal_compaction_options_set_stop_style(self.inner, style as c_int);
+        }
+    }
+}
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+#[allow(dead_code)]
+pub enum BottommostLevelCompaction {
+    /// Skip bottommost level compaction
+    Skip = 0 as isize,
```

Comment:
I suppose `u8` is more appropriate here, but I think that instead of using `as isize` for each value it is easier to use `repr(u8)` for the whole enum.